core/src/main/scalajs/sttp/client3/AbstractFetchBackend.scala

```diff
@@ -240,63 +249,104 @@ abstract class AbstractFetchBackend[F[_], S <: Streams[S], P](
       .order(original.order)
   }
 
-  protected def handleStreamBody(s: streams.BinaryStream): F[js.UndefOr[BodyInit]]
+  private def sendWebSocket[T, R >: PE](request: Request[T, R]): F[Response[T]] = {
+    val queue = new JSSimpleQueue[F, WebSocketEvent](webSocketTimeout)
+    val ws = new JSWebSocket(request.uri.toString)
+    ws.binaryType = BinaryType
 
-  private lazy val bodyFromResponseAs =
-    new BodyFromResponseAs[F, FetchResponse, Nothing, streams.BinaryStream] {
-      override protected def withReplayableBody(
-          response: FetchResponse,
-          replayableBody: Either[Array[Byte], SttpFile]
-      ): F[FetchResponse] = {
-        val bytes = replayableBody match {
-          case Left(byteArray) => byteArray
-          case Right(file)     => throw new IllegalArgumentException("Replayable file bodies are not supported")
-        }
-        new FetchResponse(bytes.toTypedArray.asInstanceOf[BodyInit], response.asInstanceOf[ResponseInit]).unit
-      }
+    ws.onopen = (_: Event) => queue.offer(WebSocketEvent.Open())
+    ws.onmessage = (event: MessageEvent) => queue.offer(toWebSocketEvent(event))
+    ws.onerror = (_: Event) => queue.offer(WebSocketEvent.Error(new WebSocketErrorException))
```

Comment:
can't we extract some more information from the event about the error?

Would be best to just throw the original exception. The `WebSocketErrorException` is used only here, so I'd define it locally, or just use `RuntimeException`

Reply:
There's no info about error in `Event` object, I will use `RuntimeException